components/sup/src/manager/service/mod.rs

```diff
@@ -279,64 +264,67 @@ impl Service {
     pub fn tick(&mut self, census_ring: &CensusRing) -> bool {
         if !self.initialized {
             if !self.all_bindings_present(census_ring) {
-                outputln!(preamble self.service_group, "Waiting to initialize service.");
+                outputln!(preamble self.service_group, "Waiting for service binds...");
                 return false;
             }
         }
-        let svc_cfg_updated = self.update_configuration(census_ring);
+
+        let svc_updated = self.update_templates(census_ring);
+        if self.update_service_files(census_ring) {
+            self.file_updated();
+        }
 
         match self.topology {
             Topology::Standalone => {
-                self.execute_hooks();
+                self.execute_hooks(census_ring);
```

Comment:
This is actually how you're recommended to do things in Rust since you can't represent two members of a struct which hold a reference on each other without using runtime borrow checking. It's not something you often see in OO langauges but you do see this pattern in functional